spec/unit/lib/temporal_spec.rb

```diff
@@ -362,6 +362,99 @@ class TestStartWorkflow < Temporal::Workflow
         end
       end
     end
+
+    describe '.await_workflow_result' do
+      class NamespacedWorkflow < Temporal::Workflow
+        namespace 'some-namespace'
+        task_queue 'some-task-queue'
+      end
+
+      it 'looks up history in the correct namespace for namespaced workflows' do
+        completed_event = Fabricate(:workflow_completed_event, result: nil)
+        response = Fabricate(:workflow_execution_history, events: [completed_event])
+
+        workflow_id = 'dummy_workflow_id'
+        run_id = 'dummy_run_id'
+        expect(client)
+          .to receive(:get_workflow_execution_history)
+          .with(
+            namespace: 'some-namespace',
+            workflow_id: workflow_id,
+            run_id: run_id,
+            wait_for_new_event: true,
+            event_type: :close,
+          )
+          .and_return(response)
+
+        Temporal.await_workflow_result(
+          NamespacedWorkflow,
+          workflow_id: workflow_id,
+          run_id: run_id,
+        )
+      end
+
+      [
+        {type: 'hash', expected_result: { 'key' => 'value' }},
+        {type: 'integer', expected_result: 5},
+        {type: 'nil', expected_result: nil},
+        {type: 'string', expected_result: 'a result'},
+      ].each do |type:, expected_result:|
+        it "completes and returns a #{type}" do
+          payload = Temporal::Api::Common::V1::Payloads.new(
+            payloads: [
+              Temporal::Client::Converter::Payload::JSON.new.to_payload(expected_result)
```

Comment:
You're forcing JSON converter here, however things such as `nil` and `string` are using other converters